gwcli/gateway.py

```diff
@@ -410,6 +410,56 @@ def ui_command_info(self):
                 console_message("- {} : {}".format(dict_key,
                                                    self.control_values[dict_key]))
 
+    def ui_command_reconfigure(self, attribute, value):
+        """
+        The reconfigure command allows you to tune various gatway attributes.
+        An empty value for an attribute resets the lun attribute to its
+        default.
+        attribute : attribute to reconfigure. supported attributes:
+          - cmdsn_depth : integer
+          - dataout_timeout : integer
+          - nopin_response_timeout : integer
+          - nopin_timeout : integer
+          - immediate_data : [Yes|No]
+          - initial_r2t : [Yes|No]
+          - first_burst_length : integer
+          - max_burst_length : integer
+          - max_outstanding_r2t : integer
+          - max_recv_data_segment_length : integer
+          - max_xmit_data_segment_length : integer
+        value     : value of the attribute to reconfigure
+        e.g.
+        set cmdsn_depth
+          - reconfigure attribute=cmdsn_depth value=128
+        reset cmdsn_depth
+          - reconfigure attribute=cmdsn_depth value=
+        """
+        if not attribute in settings.Settings.GATEWAY_SETTINGS:
+            self.logger.error("supported attributes: {}".format(",".join(
+                sorted(settings.Settings.GATEWAY_SETTINGS))))
+            return
+
+        # Issue the api request for the resize
+        gateways_api = ('{}://127.0.0.1:{}/api/'
+                        'target/{}'.format(self.http_mode,
+                                           settings.config.api_port,
+                                           self.target_iqn))
+
+        controls = {attribute: value}
+        controls_json = json.dumps(controls)
+        api_vars = {'mode': 'reconfigure', 'controls': controls_json}
+
+        self.logger.debug("Issuing reconfigure request: controls={}".format(controls_json))
+        api = APIRequest(gateways_api, data=api_vars)
+        api.put()
+
+        if api.response.status_code == 200:
+            self.logger.info('ok')
+        else:
+            self.logger.error("Failed to reconfigure : "
+                              "{}".format(response_message(api.response,
+                                                           self.logger)))
```

Comment:
I am not sure where yet, but I think we need a _set_controls/_refresh_control_values call somewhere around this time, so if the info command is run after the reconfigure then you see the updated values. Or maybe _get_controls should do a _refresh_control_values, so Target.refresh() calls update those values.

I noticed for disk reconfigure operations we do it in the reconfigure function when successful which I think is in a similar place as here.

Reply:
Updated

Reply:
Updated